// src/components/Hero.jsx
import { useState, useEffect } from "react";
import { G } from "../theme";

const ROLES = ["IT Student", "Systems Thinker", "Web Developer", "Tech Futurist"];

function TypingEffect({ words }) {
  const [display,  setDisplay]  = useState("");
  const [wordIdx,  setWordIdx]  = useState(0);
  const [charIdx,  setCharIdx]  = useState(0);
  const [deleting, setDeleting] = useState(false);

  useEffect(() => {
    const word  = words[wordIdx];
    const speed = deleting ? 50 : 100;
    const timer = setTimeout(() => {
      if (!deleting) {
        setDisplay(word.slice(0, charIdx + 1));
        if (charIdx + 1 === word.length) setTimeout(() => setDeleting(true), 2000);
        else setCharIdx(c => c + 1);
      } else {
        setDisplay(word.slice(0, charIdx - 1));
        if (charIdx - 1 === 0) { setDeleting(false); setWordIdx(i => (i + 1) % words.length); setCharIdx(0); }
        else setCharIdx(c => c - 1);
      }
    }, speed);
    return () => clearTimeout(timer);
  }, [charIdx, deleting, wordIdx, words]);

  return (
    <span>
      {display}
      <span style={{ display: "inline-block", width: 3, height: "0.85em", background: G.accent, marginLeft: 2, verticalAlign: "middle", animation: "fadeIn 0.8s ease-in-out infinite alternate" }} />
    </span>
  );
}

function Hero() {
  return (
    <section id="home" style={{ minHeight: "100vh", background: G.bg, position: "relative", overflow: "hidden", display: "flex", flexDirection: "column", justifyContent: "flex-end" }}>

      {/* Large background index number */}
      <div style={{
        position:   "absolute",
        top:        "50%",
        right:      "-2vw",
        transform:  "translateY(-50%)",
        fontFamily: "'Bebas Neue', sans-serif",
        fontSize:   "clamp(16rem, 35vw, 32rem)",
        lineHeight: 1,
        color:      "transparent",
        WebkitTextStroke: `1px ${G.rule}`,
        pointerEvents: "none",
        userSelect: "none",
        zIndex:     0,
      }}>
        NM
      </div>

      {/* Top meta bar */}
      <div style={{
        position:    "absolute",
        top:         80,
        left:        0, right: 0,
        padding:     "0 5vw",
        display:     "flex",
        justifyContent: "space-between",
        alignItems:  "center",
        zIndex:      1,
      }}>
        <span className="label">Based in Riga, Latvia 🇱🇻</span>
        <span className="label">Available for opportunities</span>
      </div>

      {/* Main content — sits at bottom of hero */}
      <div className="container" style={{ position: "relative", zIndex: 1, paddingBottom: 80 }}>

        {/* Eyebrow */}
        <div style={{ display: "flex", alignItems: "center", gap: 16, marginBottom: 24, animation: "fadeUp 0.8s ease both" }}>
          <div style={{ width: 40, height: 1, background: G.accent }} />
          <span className="label">Portfolio 2025</span>
        </div>

        {/* Giant name */}
        <h1 className="display" style={{ color: G.ink, animation: "fadeUp 0.8s ease 0.1s both", marginBottom: 0 }}>
          NICKSON
        </h1>
        <h1 className="display" style={{
          color:     "transparent",
          WebkitTextStroke: `2px ${G.ink}`,
          animation: "fadeUp 0.8s ease 0.2s both",
          marginBottom: 32,
        }}>
          MUGERWA
        </h1>

        {/* Role + description row */}
        <div style={{
          display:       "grid",
          gridTemplateColumns: "1fr 1fr",
          gap:           60,
          alignItems:    "end",
          animation:     "fadeUp 0.8s ease 0.3s both",
        }} className="two-col">

          {/* Typing role */}
          <div>
            <div style={{
              fontFamily: "'DM Serif Display', serif",
              fontSize:   "clamp(1.2rem, 2.5vw, 1.6rem)",
              color:      G.inkMid,
              marginBottom: 20,
            }}>
              <TypingEffect words={ROLES} />
            </div>
            <p style={{ fontFamily: "'Manrope'", fontSize: "1rem", color: G.inkMid, lineHeight: 1.8, maxWidth: 440, fontWeight: 300 }}>
              Driven by the conviction that <span style={{ color: G.accent, fontWeight: 600 }}>technology shapes tomorrow</span> — I build systems that connect people and solve real problems.
            </p>
          </div>

          {/* CTA + stats */}
          <div style={{ display: "flex", flexDirection: "column", gap: 32, alignItems: "flex-start" }}>
            <div style={{ display: "flex", gap: 12 }}>
              <a href="#projects"
                style={{
                  background:     G.ink,
                  color:          G.inkInverse,
                  fontFamily:     "'DM Mono', monospace",
                  fontSize:       11,
                  letterSpacing:  "0.12em",
                  textTransform:  "uppercase",
                  padding:        "14px 28px",
                  textDecoration: "none",
                  transition:     "background 0.2s",
                }}
                onMouseEnter={e => e.currentTarget.style.background = G.accent}
                onMouseLeave={e => e.currentTarget.style.background = G.ink}
              >
                View Work →
              </a>
              <a href="#contact"
                style={{
                  background:     "transparent",
                  color:          G.ink,
                  fontFamily:     "'DM Mono', monospace",
                  fontSize:       11,
                  letterSpacing:  "0.12em",
                  textTransform:  "uppercase",
                  padding:        "14px 28px",
                  textDecoration: "none",
                  border:         `1px solid ${G.rule}`,
                  transition:     "all 0.2s",
                }}
                onMouseEnter={e => { e.currentTarget.style.borderColor = G.ink; e.currentTarget.style.background = G.surface; }}
                onMouseLeave={e => { e.currentTarget.style.borderColor = G.rule; e.currentTarget.style.background = "transparent"; }}
              >
                Contact
              </a>
            </div>

            {/* Quick stats */}
            <div style={{ display: "flex", gap: 40 }}>
              {[
                { n: "6+",  label: "Projects" },
                { n: "3+",  label: "Years Learning" },
                { n: "10+", label: "Technologies" },
              ].map(s => (
                <div key={s.label}>
                  <div style={{ fontFamily: "'Bebas Neue'", fontSize: "2.2rem", color: G.accent, lineHeight: 1 }}>{s.n}</div>
                  <div className="label" style={{ marginTop: 4 }}>{s.label}</div>
                </div>
              ))}
            </div>
          </div>
        </div>
      </div>

      {/* Bottom rule */}
      <div className="rule" />
    </section>
  );
}

export default Hero;
