// src/theme.js
// ─── THEME ────────────────────────────────────────────────────────────────────
// Editorial / Brutalist aesthetic — stark, architectural, high contrast.
// Inspired by Swiss design, Bloomberg terminals, and high-end agency sites.

export const G = {
  // Base
  bg:          "#F5F2ED",      // warm off-white, like architectural paper
  bgDark:      "#0F0F0F",      // near black sections
  surface:     "#EEEBE4",      // slightly darker paper
  surfaceDark: "#1A1A1A",      // dark card surface

  // Typography
  ink:         "#0F0F0F",      // primary text — near black
  inkMid:      "#5C5C5C",      // secondary text
  inkLight:    "#A8A8A8",      // muted text
  inkInverse:  "#F5F2ED",      // text on dark bg

  // Accent — ONE bold color
  accent:      "#C84B31",      // deep burnt orange/red — confident, memorable
  accentLight: "#E8603E",      // lighter hover state

  // Borders
  rule:        "#D4CFC7",      // light divider line
  ruleDark:    "#2A2A2A",      // dark divider line

  // Functional
  white:       "#FFFFFF",
  black:       "#000000",
};

export default G;
